beanie/odm/fields.py

```diff
@@ -66,13 +68,32 @@
     from beanie.odm.documents import DocType
 
 
-def Indexed(typ, index_type=ASCENDING, **kwargs):
+@dataclass(frozen=True)
+class IndexedAnnotation:
+    _indexed: Tuple[int, Dict[str, Any]]
+
+
+def Indexed(typ=None, index_type=ASCENDING, **kwargs):
```

Comment:
Maybe the name could be `IndexedField` or `IdxField`?

Reply:
My 2¢:
-1 on introducing a new name, `Indexed` is fine. 
+1 for emitting a deprecation warning `if typ is not None` (at least for Python 3.8+/Pydantic v2; not sure if `Annotated` is supported for Pydantic v1).

Reply:
In pydantic v1, annotated is supported. It is even used for [discriminating unions](https://docs.pydantic.dev/1.10/usage/types/#nested-discriminated-unions)

Reply:
> If we don't want to keep the same interface for both ways of defining indexes

It is better to keep both interfaces. Otherwise this breaking change will affect too many users.

Reply:
Yeah, I meant keeping the old one, but making a different one for the new way. Anyways, that's your call, I'm glad you liked it!